### PR TITLE
[Docs] Add V1.4.0 to change log

### PR DESCRIPTION
**Pull request by jillnogold**

### CODE REVIEW

File: docs/change-log/index.md [hunk 1]
```diff
@@ -546,29 +577,97 @@ with a drill-down to view the steps and their details. [Tech Preview]
...
+
+### Deprecated APIs, will be removed in v1.6.0
+These APIs will be removed from the v1.6.0 code. A FutureWarning appears if you try to use them in v1.6.0.
+| Deprecated / to be removed                       | Use instead                                   |
+| ------------------------------------------------ | --------------------------------------------- |
+| clear_context() |  |
+
+
+
+### Deprecated CLI, will be removed in v1.5.0
```
Comment (alonmr) [inline docs/change-log/index.md:671]: As I see it this section should be above the "will be removed in v1.6.0" one

File: docs/change-log/index.md [hunk 2]
```diff
@@ -546,29 +577,97 @@ with a drill-down to view the steps and their details. [Tech Preview]
...
- 
+| ML-3731 | When trying to identify a failed step in a workflow with `mlrun.get_run_db().list_pipelines('project-name')`, the returned error is `None`. | To see the error, use `mlrun.db.get_pipelines()` instead. |
+| ML-3743 | Setting AWS credentials as project secret cause a build failure on EKS configured with ECR. | When using an ECR as the external container registry, make sure that the project secrets AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY have read/write access to ECR, 
+as described in the [platform documentation](https://www.iguazio.com/docs/latest-release/services/app-services/docker-registry/#create-off-cluster-registry) 
 
 ## Deprecations
 
     
-| In   | ID |Description                                                          |
+| In   | ID |Description                                  ]]                        |
```
Comment (alonmr) [inline docs/change-log/index.md:603]: is the `]]` on purpose?

**Review by alonmr — COMMENTED** [review]
A few questions/suggestions

File: docs/change-log/index.md [hunk 3]
```diff
@@ -1,25 +1,106 @@
...
+# But passing it through a parameter won't work as the string value is now a store path and not a target path:
+run2 = func2.run(..., params={"artifact": run1.outputs["my_artifact"]})
+```
+
+
+### Closed issues
+
+| ID   | Description                                                    |
+| --- | ----------------------------------------------------------------- |
+| ML-1787 | Optimized distribution of load between chief and workers so that heavy loads do not cause restart of kubelet. [View in Git](https://github.com/mlrun/mlrun/pull/). |
```
[the remark this answers is not in the document]
Comment (jillnogold) [reply docs/change-log/index.md:79]: I am reviewing again now and cleaning up these loose ends

File: docs/change-log/index.md (hunk 1 above)
Comment (jillnogold) [reply docs/change-log/index.md:671]: I grouped them by API/CLI and not release. Both work - but switched them

File: docs/store/artifacts.md [hunk 4]
```diff
@@ -44,9 +42,11 @@ You can download the artifact. You can also tag and remove tags from artifacts u
 
 ## Artifact path
 
-Any path that is supported by MLRun can be used to store artifacts. However, only artifacts that are stored in paths that are system-configured as "allowed" in the MLRun service are visible in the UI. These are:
+Any path that is supported by MLRun can be used to store artifacts. However, only artifacts that are stored in paths that are 
+system-configured as "allowed" in the MLRun service are visible in the UI. These are:
 - MLRun < 1.2: The allowed paths include only v3io paths
-- MLRun 1.2 and higher: Allows cloud storage paths &mdash; `v3io://`, `s3://`, `az://`, `gcs://`, `gs:// ` 
+- MLRun 1.2 and higher: Allows cloud storage paths &mdash; `v3io://`, `s3://`, `az://`, `gcs://`, `gs:// `. `http` paths are not visible
```
[the remark this answers is not in the document]
Comment (jillnogold) [reply docs/store/artifacts.md:48]: updated

File: docs/change-log/index.md (hunk 2 above)
Comment (jillnogold) [reply docs/change-log/index.md:603]: deleted

File: docs/change-log/index.md [hunk 5]
```diff
@@ -545,30 +579,115 @@ with a drill-down to view the steps and their details. [Tech Preview]
...
+| ------------------------------------------------ | --------------------------------------------- |
+| MLRunProjectclear_context() | This method deletes all files and clears the context directory or subpath (if defined). This method can produce unexpected outcomes and is not recommended.  |
+| MLRunProject object legacy parameters | metadata and spec instead |
+| BaseRuntime.with_commands and KubejobRuntime.build_config'verify_base_image' param | 'prepare_image_for_deploy' |
+| run_local | function.run(local=True) |
+
+### REST APIs deprecated in v1.4.0, will be removed from v1.6.0 code
+| Deprecated                        | Use instead                                   |
+| ------------------------------------------------ | --------------------------------------------- |
+| http request POST /artifact/{project}/{uid}/{key:path} | /projects/{project}/artifacts/{uid}/{key:path} instead |
```
Comment (alonmr) [inline docs/change-log/index.md:634]: Saying "REST APIs"  in the title is enough :)
```suggestion
| POST /artifact/{project}/{uid}/{key:path} | /projects/{project}/artifacts/{uid}/{key:path} instead |
```

File: docs/change-log/index.md [hunk 6]
```diff
@@ -545,30 +579,115 @@ with a drill-down to view the steps and their details. [Tech Preview]
...
+## Deprecated APIs and CLI
+
+### APIs deprecated in v1.4.0, will be removed from v1.6.0 code
+These APIs will be removed from the v1.6.0 code. A FutureWarning appears if you try to use them in v1.4.0 and higher.
+
+| Deprecated / to be removed                       | Use instead                                   |
+| ------------------------------------------------ | --------------------------------------------- |
+| MLRunProjectclear_context() | This method deletes all files and clears the context directory or subpath (if defined). This method can produce unexpected outcomes and is not recommended.  |
+| MLRunProject object legacy parameters | metadata and spec instead |
+| BaseRuntime.with_commands and KubejobRuntime.build_config'verify_base_image' param | 'prepare_image_for_deploy' |
```
Comment (alonmr) [inline docs/change-log/index.md:628]:
```suggestion
| BaseRuntime.with_commands and KubejobRuntime.build_config 'verify_base_image' param | 'prepare_image_for_deploy' |
```

**Review by alonmr — COMMENTED** [review]
minor

File: docs/change-log/index.md [hunk 7]
```diff
@@ -59,12 +154,40 @@ New sections describing [Git best practices](../projects/git-best-practices.html
...
-See instructions in [Set up your environment](../install/remote.html).
-
-MLRun is pre-installed in CE Jupyter.
-
+python 3.7 have the suffix: `-py37`. The correct version is automatically chosen for the built-in MLRun images according to the Python version of the MLRun client (for example, a 3.7 Jupyter gets the `-py37` images).
+
+MLRun v1.3.x maintains support for mlrun base images that are based on a python 3.7 environment. To differentiate between the images, the images based on
+python 3.7 have the suffix: `-py37`. The correct version is automatically chosen for the built-in MLRun images according to the Python version of the MLRun client (for example, a 3.7 Jupyter gets the `-py37` images).
+
+For a Python 3.9 environment see [Set up a Python 3.7 client environment](../install/remote.html#set-up-a-python-3-9-client-environment).
```
Comment (alonmr) [inline docs/change-log/index.md:162]:
```suggestion
For a Python 3.9 environment see [Set up a Python 3.9 client environment](../install/remote.html#set-up-a-python-3-9-client-environment).
```

File: docs/change-log/index.md [hunk 8]
```diff
@@ -545,30 +620,115 @@ with a drill-down to view the steps and their details. [Tech Preview]
...
+| ML-3605 | v1.5.0 | Model Monitoring:  Most of the charts and KPIs in Grafana are now based on the data store target instead of the MLRun API. It is recommended to update the model monitoring dashboards since the old dashboards won't be supported. |
+
+## Deprecated APIs and CLI
+
+### APIs deprecated in v1.4.0, will be removed from v1.6.0 code
+These APIs will be removed from the v1.6.0 code. A FutureWarning appears if you try to use them in v1.4.0 and higher.
+
+| Deprecated / to be removed                       | Use instead                                   |
+| ------------------------------------------------ | --------------------------------------------- |
+| MLRunProjectclear_context() | This method deletes all files and clears the context directory or subpath (if defined). This method can produce unexpected outcomes and is not recommended.  |
```
Comment (alonmr) [inline docs/change-log/index.md:667]:
```suggestion
| MLRunProject.clear_context() | This method deletes all files and clears the context directory or subpath (if defined). This method can produce unexpected outcomes and is not recommended.  |
```

File: docs/change-log/index.md [hunk 9]
```diff
@@ -545,30 +620,115 @@ with a drill-down to view the steps and their details. [Tech Preview]
...
+| Deprecated                        | Use instead                                   |
+| ------------------------------------------------ | --------------------------------------------- |
+| POST /artifact/{project}/{uid}/{key:path} | /projects/{project}/artifacts/{uid}/{key:path} instead |
+| http request GET /projects/{project}/artifact/{key:path} | /projects/{project}/artifacts/{key:path} instead |
+| http request DELETE /artifact/{project}/{uid} | /projects/{project}/artifacts/{uid} instead |
+| http request GET /artifacts | /projects/{project}/artifacts instead |
+| http request DELETE /artifacts  | /projects/{project}/artifacts instead |
+| http request POST /func/{project}/{name}  | /projects/{project}/functions/{name} instead |
+| http request GET /func/{project}/{name}  | /projects/{project}/functions/{name} instead |
+| http request GET /funcs  | /projects/{project}/functions instead |
```
Comment (alonmr) [inline docs/change-log/index.md:682]:
```suggestion
| GET /projects/{project}/artifact/{key:path} | /projects/{project}/artifacts/{key:path} instead |
| DELETE /artifact/{project}/{uid} | /projects/{project}/artifacts/{uid} instead |
| GET /artifacts | /projects/{project}/artifacts instead |
| DELETE /artifacts  | /projects/{project}/artifacts instead |
| POST /func/{project}/{name}  | /projects/{project}/functions/{name} instead |
| GET /func/{project}/{name}  | /projects/{project}/functions/{name} instead |
| GET /funcs  | /projects/{project}/functions instead |
```

**Review by alonmr — CHANGES_REQUESTED** [review]
A few typos

**Review by alonmr — APPROVED** [review]

**Pull request merged**